### PR TITLE
Use interface instead of *testing.T

### LINKED ISSUE #21: use an interface instead of *testing.T
It's unfortunate that [`*testing.T`](https://godoc.org/testing#T) not an interface.

I want to use this library with Ginkgo and [documentation](https://onsi.github.io/ginkgo/#third-party-integrations) suggests that I should create an interface. Any other ideas are welcome! 👍

### PR DESCRIPTION
**Pull request by errordeveloper**
Fixes #21

### CODE REVIEW

File: test.go [hunk 1]
```diff
@@ -31,21 +33,21 @@ type Test struct {
...
-	t            *testing.T
+	t            testing.T
 	logger       logger.Logger
 	inError      bool
 	namespaces   []string // List of namespaces created by the test
 	cleanUpFns   []finalizer
 }
 
-func testName(t *testing.T) string {
+func testName(t T) string {
```
Comment (errordeveloper) [inline test.go]: not sure if this is still needed, as we have an interface and not a pointer

File: test.go [hunk 2]
```diff
@@ -31,21 +33,21 @@ type Test struct {
...
 	if t != nil {
 		return t.Name()
 	}
 	return "undefined-test"
 }
 
-func testLogger(l logger.Logger, t *testing.T) logger.Logger {
+func testLogger(l logger.Logger, t T) logger.Logger {
 	if t != nil {
```
Comment (errordeveloper) [inline test.go]: also these `t != nil` checks don't seem to be of much user – wdyt?

File: test.go [hunk 3]
```diff
@@ -31,21 +32,21 @@ type Test struct {
...
+func testName(t testing.T) string {
 	if t != nil {
 		return t.Name()
 	}
 	return "undefined-test"
 }
 
-func testLogger(l logger.Logger, t *testing.T) logger.Logger {
+func testLogger(l logger.Logger, t testing.T) logger.Logger {
 	if t != nil {
```
[the remark this answers is not in the document]
Comment (dlespiau) [reply test.go]: Yes we can remove the `nil` interface tests!

**Comment by errordeveloper** [comment]
So looks like there is little use to `PrintfLogger` now...

**Comment by errordeveloper** [comment]
> So looks like there is little use to `PrintfLogger` now...

Well, not true, had to revert the `l.t != nil` code in `test_logger.go` to fix the panic in CI. Without that, logger needs to be constructed explicitly, which breaks backwards compatibility.

**Pull request merged**